### DISCUSSION TITLE
Code coverage

### DISCUSSION THREAD

**Discussion by freddydk** [microsoft/AL-Go#115] [Ideas]
Add automatic code coverage reporting when running tests
Report code coverage using standard formats
Add graphical code coverage viewer

**Comment by StefanMaron** [comment]
Yes Please!

**Reply by gntpet** [reply]
I wonder if there's any plan for code coverage still?

**Reply by frottke** [reply]
> I wonder if there's any plan for code coverage still?

We’re waiting too. Would be quite helpful.

**Comment by SchulzOli** [comment]
would love to use it. Any Update on that?

**Reply by frottke** [reply]
we would also love to use it. 👍

**Reply by spetersenms** [reply]
Good news! I am currently working on code coverage. Here is a sneak peak:
<img width="717" height="1036" alt="image" src="https://github.com/user-attachments/assets/a6c28a4c-063e-4de4-9bbd-2ad437e3c016" />

**Reply by frottke** [reply]
@spetersenms awesome!
Would it be possible to add a setting in the AL-Go settings that defines a minimum required code coverage percentage?
This would help enforce consistent test quality. By defining a threshold, the pipeline can prevent changes with insufficient test coverage from being merged.
